### PR TITLE
DDF-2819 Add maven-version-validation-plugin to root pom.xml

### PR DESCRIPTION
**Pull request by mweser**
#### What does this PR do?
Enables a ddf-support maven plugin which scans the project at build for any package.json files which incorrectly use version ranges for dependencies.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@ahoffer @emanns95 @AzGoalie @vinamartin 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Build](https://github.com/orgs/codice/teams/build)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@pklinef

#### How should this be tested? (List steps with links to updated documentation)
Will need to pull in ddf-support PR as well (https://github.com/codice/ddf-support/pull/38) for the actual plugin, and then run `mvn verify` in DDF. Successful if verify errors out with a MojoException due to incorrect version range symbols. 

#### Any background context you want to provide?
Version ranges expose the codebase to potential vulnerabilities, unlike sticking to exact, known versions of dependencies. 

#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/DDF-2819)

#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -582,6 +583,19 @@
                 </pluginManagement>
                 <plugins>
                     <plugin>
+                        <groupId>ddf.support</groupId>
+                        <artifactId>version-validation-plugin</artifactId>
+                        <version>2.3.7-SNAPSHOT</version>
```
Comment (emanns95) [inline pom.xml]: We should use the ddf.support.version property for the version

Comment (mweser) [inline pom.xml]: Like this: `${ddf.support.version}`?

File: pom.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
         <reports.repository.url/>
         <project.scm.id>github</project.scm.id>
         <!-- Must MANUALLY update this if the ddf/support project's version changes -->
-        <ddf.support.version>2.3.6</ddf.support.version>
+        <ddf.support.version>2.3.7-SNAPSHOT</ddf.support.version>
```
Comment (pklinef) [inline pom.xml]: @shaundmorris Do you usually move to SNAPSHOT or do we just release support and directly move to the next released version?

Comment (shaundmorris) [inline pom.xml]: @pklinef we should just release support and move to the next released version

**Comment by mweser** [comment]
build now

**Comment by mweser** [comment]
Passes full swarm build.

**Pull request merged**